gpt_index/vector_stores/opensearch.py

```diff
@@ -0,0 +1,202 @@
+"""
+Elasticsearch/Opensearch vector store.
+"""
+from gpt_index.data_structs import Node
+
+
+import json
+from typing import Any, List
+from gpt_index.vector_stores.types import (
+    NodeEmbeddingResult,
+    VectorStore,
+    VectorStoreQueryResult,
+)
+
+
+class OpensearchVectorClient:
+    """
+    Object encapsulating an Opensearch index that has vector search enabled.
+
+    If the index does not yet exist, it is created during init.
+    Therefore, The underlying index is assumed to either:
```

Comment:
wasn't too sure if this is the ideal spot in the codebase for this class. let me know if there's a better spot for it